site-src/geps/gep-713.md

```diff
@@ -0,0 +1,409 @@
+    cdn:
+      cachePolicy:
+        includeHost: true
+        includeProtocol: true
+        includeQueryString: true
+  targetRef:
+    kind: Gateway
+    name: example
+---
+kind: GKEServicePolicy
+spec:
+  default:
+    cdn:
+      cachePolicy:
+        includeQueryString: false
+  targetRef:
+    kind: HTTPRoute
+    name: example
+```
+
+### Targeting External Services
+In some cases (likely limited to mesh) we may want to apply policies to requests
+to external services. To accomplish this, implementations can choose to support
+a refernce to a virtual resource type:
+
+```yaml
+kind: RetryPolicy
+apiVersion: networking.acme.io/v1alpha1
+metadata:
+  name: foo
+spec:
+  default:
+    maxRetries: 5
+  targetRef:
+    group: networking.acme.io
+    kind: ExternalService
+    hostname: foo.com
+```
+
+### Apply Policies to Specific Matches
+In some cases we may want to target specific matches within nested objects. For
+example, it may be useful to attach policies to a specific Gateway listener or
+Route rule.
+
+Each Route rule or Gateway listener should be expanded with an optional name
+field. The target ref would be expanded with an optional sectionName field that
+could be used to refer to that specific section of the resource. It would refer
+to the following concepts on these resources:
+
+* Gateway.Listeners.Name
+* xRoute.Rules.Name
+* Service.Ports.Name
+
+```yaml
+kind: HTTPRoute
+apiVersion: networking.x-k8s.io/v1alpha1
+metadata:
+  name: http-app-1
+  labels:
+    app: foo
+spec:
+  hostnames:
+  - "foo.com"
+  rules:
+  - name: bar
+    matches:
+    - path:
+        type: Prefix
+        value: /bar
+    forwardTo:
+    - serviceName: my-service1
+      port: 8080
+---
+kind: RetryPolicy
+apiVersion: networking.acme.io/v1alpha1
+metadata:
+  name: foo
+spec:
+  maxRetries: 5
+  targetRef:
+    name: foo
+    group: networking.x-k8s.io
+    kind: HTTPRoute
+    sectionName: bar
+```
+
+This would require adding a `Name` field to Gateway listeners and Route rules:
+
+```go
+type Listener struct {
+    // Name is the name of the Listener.
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=253
+    // +optional
+    Name string `json:"name,omitempty"`
+    // ...
+}
+```
+
+```go
+type RouteRule struct {
+    // Name is the name of the Route rule.
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=253
+    // +optional
+    Name string `json:"name,omitempty"`
+    // ...
+}
+```
+
+### Kubectl Plugin
+To help improve UX and standardization, a kubectl plugin will be developed that
+will be capable of describing the computed sum of policy that applies to a given
+resource, including policies applied to parent resources.
+
+Each Policy CRD that wants to be supported by this plugin will need to follow
+the API structure defined above and add a `networking.x-k8s.io/policy: true` to
```

Comment:
```suggestion
the API structure defined above and add a `networking.x-k8s.io/policy: true` label to
```
or annotation. label would be easier to query

Reply:
woops, meant to say label here, as you mentioned for the simpler querying.

Reply:
updated now, thanks!